lib/inspec/resources/windows_firewall.rb

```diff
@@ -0,0 +1,108 @@
+require "inspec/resources/powershell"
+
+module Inspec::Resources
+  class WindowsFirewall < Inspec.resource(1)
+    name "windows_firewall"
+    supports platform: "windows"
+    desc "Check properties of the Windows Firewall for a specific profile."
+    example <<~EXAMPLE
+      describe windows_firewall("Public") do
+        it { should be_enabled }
+        its("default_inbound_action") { should_not cmp "NotConfigured" }
+        its("num_rules") { should be 19 }
+      end
+    EXAMPLE
+
+    def initialize(profile = "Public")
+      @profile = profile
+      @state = {}
+
+      load_profile_cmd = load_firewall_profile(profile)
+      cmd = inspec.powershell(load_profile_cmd)
+
+      @state = JSON.load(cmd.stdout) unless cmd.stdout.empty?
+    end
+
+    def to_s
+      "Windows Firewall (Profile #{@profile})"
+    end
+
```

Comment:
It's good to always implement an `exist?` method so auditors can always assert that the resource even exists.
Looking at [`load_profile_profile`](https://github.com/inspec/inspec/pull/4979/files#diff-9207d1cb1fd45ca62aff2e4304f273a1R76) it looks like it is assumed that the named profile will exist.

Reply:
Added the method